### PR TITLE
Issue #1439: Add default values for image field alt and title text.

### LINKED ISSUE #1439: Allow setting a default value for Alt and Title text in image fields
This is basically the same issue as https://www.drupal.org/node/807126 that was filed against the D6 branch of [ImageField](https://www.drupal.org/project/imagefield). That issue was closed as "fixed" by @quicksketch by pointing to the [ImageField Tokens](https://www.drupal.org/project/imagefield_tokens) module:

imagefield_token_settings

Now with Tokens in core, it makes sense to take advantage of them and use them where possible.

PS: should support transliteration once in core (#904)

---

PR by @BWPanda: https://github.com/backdrop/backdrop/pull/3068

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/1439

### CODE REVIEW

File: core/modules/image/image.field.inc [hunk 1]
```diff
@@ -534,8 +586,8 @@ function image_field_formatter_info() {
         'image_link' => '',
         'image_float' => '',
       ),
-	  'description' => t('Format the file as an image. The image can be displayed using an image style and can optionally be linked to the image file itself or its parent content.'),
-	  'file formatter' => array(
+	    'description' => t('Format the file as an image. The image can be displayed using an image style and can optionally be linked to the image file itself or its parent content.'),
```
Comment (klonos) [inline core/modules/image/image.field.inc]:
```suggestion
	  'description' => t('Format the file as an image. The image can be displayed using an image style and can optionally be linked to the image file itself or its parent content.'),
```

File: core/modules/image/image.field.inc [hunk 2]
```diff
@@ -534,8 +586,8 @@ function image_field_formatter_info() {
...
+	    'file formatter' => array(
```
Comment (klonos) [inline core/modules/image/image.field.inc]:
```suggestion
	  'file formatter' => array(
```

File: core/modules/image/image.field.inc (hunk 1 above)
Comment (ghost) [reply core/modules/image/image.field.inc]: Weird stuff going on with indentation. Fixed now.

File: core/modules/image/image.field.inc (hunk 2 above)
Comment (ghost) [reply core/modules/image/image.field.inc]: Weird stuff going on with indentation. Fixed now.

File: core/modules/image/image.field.inc [hunk 3]
```diff
@@ -498,24 +531,31 @@ function image_field_widget_process($element, &$form_state, $form) {
...
+  ));
   $element['alt'] = array(
     '#title' => t('Alternate text'),
     '#type' => 'textfield',
-    '#default_value' => isset($item['alt']) ? $item['alt'] : '',
-    '#description' => t('This text will be used by screen readers, search engines, or when the image cannot be loaded.'),
-    // @see http://www.gawds.org/show.php?contentid=28
+    '#default_value' => isset($item['alt']) ? $item['alt'] : $settings['alt_default_value'],
+    '#description' => t('This text will be used by screen readers, search engines, or when the image cannot be loaded. This field supports tokens.') . ' ' . $token_tree_link,
+    // @see https://web.archive.org/web/20121120150943/http://www.gawds.org/show.php?contentid=28
```
Comment (klonos) [inline core/modules/image/image.field.inc:566]: 👍

**Review by klonos — APPROVED** [review]
LGTM 👍

**Comment by quicksketch** [comment]
Getting a new sandbox.

**Pull request closed without merging**